### PR TITLE
feat: Download Sequence Variability Data as Fasta File

### LINKED ISSUE #37: Download Sequence Variability DataFrame as FASTA file
This issue tracks the following feature(s) -

- The PI's want to download the Sequence Variability Table to a separate FASTA file

### PR DESCRIPTION
**Pull request by aniketfadia96**
Hi, I have made the following changes in this PR -

- Convert the Sequence Variability DataFrame to a FASTA format string
- Add a button to download the above string in the FASTA file

I have attached a screencast for your reference - 

[Seq-Var-FASTA-file-download.webm](https://github.com/uw-ssec/neglected-diagnostics/assets/25303819/b04c2d2b-4ba2-4624-a851-9e98ccfc8a10)

Closes #37

### CODE REVIEW

File: src/app/frontend/download_data.py [hunk 1]
```diff
@@ -46,3 +48,30 @@ def download_data(database: str, ids: List[int], search_term: str) -> None:
...
+    ----------
+    df : pd.DataFrame
+        A DataFrame containing sequence variability data.
+    species : str
+        The name of the species for which the data is being processed.
+    """
+    # Convert the DataFrame to FASTA format string
+    fasta_str = seqvar_df_to_fasta(df, species)
+
+    st.write("FASTA file ready for download!")
```
Comment (lsetiawan) [inline src/app/frontend/download_data.py:70]:
```suggestion
    st.write("FASTA file is ready for download. Please click the button below to proceed.")
```

The more instructional the better for copy writing so it's clear for the user about the action that they should take 😄 I assume that the button is underneath this piece of text.

File: src/app/pages/2_Sequence-Variability.py [hunk 2]
```diff
@@ -25,9 +31,30 @@
...
+
+        # Display the sequence variability data
+        if not st.session_state[SEQVAR_DF].empty:
+            st.write("Sequence Variability Data:")
+            st.dataframe(st.session_state[SEQVAR_DF])
+
+            # Enter the species name for downloading the data
+            species = st.text_input("Enter the species name:")
+
+            if st.button("Prepare for Download"):
```
Comment (lsetiawan) [inline src/app/pages/2_Sequence-Variability.py:57]:
```suggestion
            if st.button("Prepare for download"):
```

File: src/app/pages/2_Sequence-Variability.py [hunk 3]
```diff
@@ -25,9 +31,30 @@
...
+                download_seq_var_data(st.session_state[SEQVAR_DF], species)
         else:
-            df = calculate_sequence_variability(sequences, base_sequence)
-            st.write("Sequence Variability Calculation Results:")
-            st.dataframe(df)
+            st.write("No Sequence Variability Data to Download")
```
Comment (lsetiawan) [inline src/app/pages/2_Sequence-Variability.py:60]:
```suggestion
            st.write("No sequence variability data to download")
```

Comment (lsetiawan) [reply src/app/pages/2_Sequence-Variability.py:60]: Wondering if you need to capitalize on `sequence variability` or not? I feel that it's unnecessary to do so.

Comment (aniketfadia96) [reply src/app/pages/2_Sequence-Variability.py:60]: Yes, looks better without capitalize.

**Review by lsetiawan — APPROVED** [review]

**Pull request merged**